### PR TITLE
Add APIs required for the CalDAV trashbin

### PR DESCRIPTION
**Pull request by ChristophWurst**
Required for https://github.com/nextcloud/calendar/pull/3118

* Expose new property specific to deleted calendars
* API to MOVE calendars (or anything else) from one location to another

### CODE REVIEW

**Review by tcitworld — CHANGES_REQUESTED** [review]

File: src/parser.js [hunk 1]
```diff
@@ -193,6 +193,7 @@ export default class Parser {
 		this.registerParser('{http://owncloud.org/ns}enabled', Parser.bool);
 		this.registerParser('{http://owncloud.org/ns}read-only', Parser.bool);
 		this.registerParser('{http://nextcloud.com/ns}owner-displayname', Parser.text);
+		this.registerParser('{http://nextcloud.com/ns}deleted-at', Parser.decInt);
```
[the remark this answers is not in the document]
Comment (ChristophWurst) [reply src/parser.js:196]: what other format should we use? we use timestamps quite consistently across nextcloud

[the remark this answers is not in the document]
Comment (tcitworld) [reply src/parser.js:196]: It would be nice to be readable straight away, so I'd say ISO8601?

[the remark this answers is not in the document]
Comment (ChristophWurst) [reply src/parser.js:196]: Fair enough. I'll see what I can do once the basic functionality is there.

[the remark this answers is not in the document]
Comment (ChristophWurst) [reply src/parser.js:196]: I will schedule a refactoring after the basics are done. This change also requires a touch of the server part, obviously ;)

**Review by GretaD — APPROVED** [review]

**Pull request merged**